doc/man3/OSSL_trace_enabled.pod

```diff
@@ -136,26 +152,50 @@ This will normally expand to:
  } while (0);
 
 
-C<OSSL_TRACE1()>, ... C<OSSL_TRACE9()> are one-shot macros which essentially wrap
-a single BIO_printf() into a tracing group.
+C<OSSL_TRACE()> and C<OSSL_TRACE1()>, C<OSSL_TRACE1()>, ... C<OSSL_TRACE9()> are
+so-called one-shot macros:
 
-The call OSSL_TRACEn(category, format, arg1, ..., argN) expands to:
+The macro call C<OSSL_TRACE(category, text)>, produces literal text trace output.
+
+The macro call C<OSSL_TRACEn(category, format, ...arg1, ..., argn)> produces
+printf-style trace output with n format field arguments (n=1,...9).
+It expands to:
+
+ OSSL_TRACE_BEGIN(category) {
+   BIO_printf(trc_out, format, arg1, ..., argN)
+ } OSSL_TRACE_END(category)
 
-  OSSL_TRACE_BEGIN(category) {
-    BIO_printf(trc_out, format, arg1, ..., argN)
-  } OSSL_TRACE_END(category)
+Internally, all one-shot macros are implemented using a generic C<OSSL_TRACEV()>
+macro, since C90 does not support variadic macros. This helper macro has a rather
+weird synopsis and is not intended to be used directly.
+
+The C<OSSL_TRACE_ENABLED(category)> macro can be used to conditionally execute
+some code only if a specific trace category is enabled.
+In some situations this is simpler than entering a trace section using
+C<OSSL_TRACE_BEGIN(category)> and C<OSSL_TRACE_END(category)>.
+For example, the code
+
+ if (OSSL_TRACE_ENABLED(TLS)) {
+     ...
+ }
+
+expands to
+
+ if (OSSL_trace_enabled(OSSL_TRACE_CATEGORY_TLS) {
+     ...
+ }
 
 =head1 NOTES
 
 It is advisable to always check that a trace type is enabled with
 OSSL_trace_enabled() before generating any output, for example:
 
-    if (OSSL_trace_enabled(OSSL_TRACE_CATEGORY_TLS)) {
-        BIO *trace = OSSL_trace_begin(OSSL_TRACE_CATEGORY_TLS);
-        BIO_printf(trace, "FOO %d\n", somevalue);
-        BIO_dump(trace, somememory, somememory_l);
-        OSSL_trace_end(OSSL_TRACE_CATEGORY_TLS, trace);
-    }
+     if (OSSL_trace_enabled(OSSL_TRACE_CATEGORY_TLS)) {
```

Comment:
_1+4 (one space for perlpod, 4 spaces for C)_

Reply:
I just noticed that the entire NOTES section needs an update. I'll put the pull request in WIP until it's done...